src/ACEtk/block/SecondaryParticleLocatorBlock.hpp

```diff
@@ -0,0 +1,189 @@
+#ifndef NJOY_ACETK_BLOCK_SECONDARYPARTICLELOCATORBLOCK
+#define NJOY_ACETK_BLOCK_SECONDARYPARTICLELOCATORBLOCK
+
+// system includes
+
+// other includes
+#include "ACEtk/block/details/Base.hpp"
+
+namespace njoy {
+namespace ACEtk {
+namespace block {
+
+/**
+ *  @class
+ *  @brief The continuous energy IXS block with the secondary particle locators
+ *
+ *  The SecondaryParticleLocatorBlock class contains 10 locators for each
+ *  secondary particle type as given on the PTYPE block.
+ *
+ *  The number of available particle types is stored in NXS(7).
+ */
+class SecondaryParticleLocatorBlock : protected details::Base {
+
+  /* fields */
+  unsigned int ntype_;
+
+  /* auxiliary functions */
+  #include "ACEtk/block/SecondaryParticleLocatorBlock/src/generateXSS.hpp"
+  #include "ACEtk/block/SecondaryParticleLocatorBlock/src/verifySize.hpp"
+
+public:
+
+  /* constructor */
+  #include "ACEtk/block/SecondaryParticleLocatorBlock/src/ctor.hpp"
+
+  /* methods */
+
+  /**
+   *  @brief Return the number of secondary particle types (excluding the
+   *         projectile)
+   */
+  unsigned int NTYPE() const { return this->ntype_; }
+
+  /**
+   *  @brief Return the number of secondary particle types (excluding the
+   *         projectile)
+   */
+  unsigned int numberAdditionalSecondaryParticleTypes() const {
+
+    return this->NTYPE();
+  }
+
+  /**
+   *  @brief Return the locator for a secondary particle index and block index
+   *
+   *  @param[in] particle    the secondary particle index (one-based)
+   *  @param[in] index       the block index (one-based)
+   */
+  auto LLOC( std::size_t particle, std::size_t index ) const {
+
+    return this->XSS( ( particle - 1 ) * 10 + index );
+  }
+
+  /**
+   *  @brief Return the locator for a secondary particle index and block index
+   *
+   *  @param[in] particle    the secondary particle index (one-based)
+   *  @param[in] index       the block index (one-based)
+   */
+  auto locator( std::size_t particle, std::size_t index ) const {
+
+    return this->XSS( ( particle - 1 ) * 10 + index );
```

Comment:
In most of the rest of the codebase, when things are simply renamed, they call another function to have a single point of truth. 

Is there a reason `locator` doesn't just call `LLOC`?

Reply:
Yup, that should have been a call to the previous one. I seem to have fixed that at a later point in time.